Components/Core/Inc/Timer.hpp

```diff
@@ -39,7 +49,11 @@ class Timer
 
 protected:
 	TimerHandle_t rtTimerHandle;
+	CurrentState timerState;
+	TickType_t remainingTime;
```

Comment:
If RTOS is keeping track of this, we shouldn't have to keep track of this. But if it makes it more efficient, keep it, just remember it won't always be consistent (you must update this every time you do any operation with it).

Also, please make this a uint32_t with milliseconds and don't use TickType_t. Append the type of the variable in somewhat obscure things like this, eg remainingTimeMs

I recommend writing a GetRemainingTime() function